File: rock.java
import java.util.Scanner;
//import java.util.io.*;
import java.util.Random;

class RockPaperScissor {

	static void requestPlay(String[] weapons){
		String display = "Please select";
		/*for (int i=0;i<weapons.length;i++){*/
		int i=0;
		for (String weapon: weapons){
			i++;
			display += " " + i + ". " + weapon;
		};
		System.out.println(display);
	}
	
	public static void main(String[ ] args) {
		RockPaperScissor rockPaperScissor = new RockPaperScissor();
		rockPaperScissor.run();
	}
	public void run(){
		int userWeapon;
		/*BufferedReader userInput = 
		 new BufferedReader( new InputStreamReader(System.in));*/
		 
		Scanner userInput = new Scanner(System.in);
		Random rand = new Random();
		int computerWeapon;
		//Final declares a constant
		final String[] weaponList = {"Rock","Scissors","Paper"};
		//do{
			requestPlay(weaponList);
	  		userWeapon = userInput.nextInt()-1;
	  		computerWeapon = rand.nextInt(3);
			System.out.println("Users selected "+ weaponList[userWeapon] + " Computer selected " + weaponList[computerWeapon]);
			if (userWeapon == computerWeapon){
				System.out.println("Draw");
			} else if ((userWeapon + 1)%3 == computerWeapon){
				System.out.println("You won");
			} else if ((computerWeapon+1)%3 == userWeapon) {
				System.out.println("Computer won");
			}
		//} while (userWeapon< 4);
	}
}

File: rock.cs
using System;
/*using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
*/
namespace SoloLearn
{
    class rock
    {
        static string generateRequest(String[] weapons)
        {
            String request = "Please select";
            int i = 0;
            foreach (string weapon in weapons)
            {
                i++;
                request += " " + i + ". " + weapon;
            }
            request += " 4. Exit";
            return request;
        }
        static int requestPlay(String[] weapons)
        {
            String request;
            request = generateRequest(weapons);
            Console.WriteLine(request);

            int userWeapon = Convert.ToInt32(Console.ReadLine()) - 1;

            return userWeapon;
        }
        static void displayWinner(string winner)
        {
            Console.WriteLine(winner);
        }
        static string determineWinner(String[] weaponList, int userWeapon, int computerWeapon)
        {
            String winner;
            if (userWeapon == computerWeapon)
            {
                winner = "Draw both selected " + weaponList[computerWeapon];
            }
            else if ((userWeapon + 1) % weaponList.Length == computerWeapon)
            {
                winner = "You win and beat the computer's " + weaponList[computerWeapon];
            }
            else if ((computerWeapon + 1) % weaponList.Length == userWeapon)
            {
                winner = "Computer wins with " + weaponList[computerWeapon];
            }
            else
            {
                winner = "Please select 1. Rock, 2. Scissors or 3. Paper";
            }

            return winner;

        }

        static void play(String[] weaponList)
        {
            int userWeapon;
            userWeapon = requestPlay(weaponList);
            while (userWeapon < weaponList.Length)
            {
                int computerWeapon;
                string winner;
                Random random = new Random();
                computerWeapon = random.Next(0, weaponList.Length -1);

                winner = determineWinner(weaponList, userWeapon, computerWeapon);
                displayWinner(winner);
                userWeapon = requestPlay(weaponList);
            }
        }

        static void Main(string[] args)
        {
            var weaponList = new String[] { "Rock", "Scissors", "Paper" };
            play(weaponList);
        }
    }
}